plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1090 @@
+            """create groups defined by lxd.yml or defaultvalues
+
+            create groups defined by lxd.yml or defaultvalues
+            supportetd:
+                * 'location'
+                * 'pattern'
+                * 'network_range'
+                * 'os'
+                * 'release'
+                * 'profile'
+                * 'vlanid'
+
+            Args:
+                str(group_name): Group name
+            Kwargs:
+                None
+            Raises:
+                None
+            Returns:
+                None"""
+
+            # Due to the compatibility with python 2 no use of map
+            if self.groups[group_name].get('type') == 'location':
+                self.build_inventory_groups_location(group_name)
+            elif self.groups[group_name].get('type') == 'pattern':
+                self.build_inventory_groups_pattern(group_name)
+            elif self.groups[group_name].get('type') == 'network_range':
+                self.build_inventory_groups_network_range(group_name)
+            elif self.groups[group_name].get('type') == 'os':
+                self.build_inventory_groups_os(group_name)
+            elif self.groups[group_name].get('type') == 'release':
+                self.build_inventory_groups_release(group_name)
+            elif self.groups[group_name].get('type') == 'profile':
+                self.build_inventory_groups_profile(group_name)
+            elif self.groups[group_name].get('type') == 'vlanid':
+                self.build_inventory_groups_vlanid(group_name)
+            else:
+                raise AnsibleParserError('Unknown group type: {0}'.format(to_native(group_name)))
+
+        for group_name in self.groups.keys():
+            if not group_name.isalnum():
+                raise AnsibleParserError('Invalid character(s) in groupname: {0}'.format(to_native(group_name)))
+            group_type(group_name)
+
+    def build_inventory(self):
+        """Build dynamic inventory
+
+        Build the dynamic inventory.
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            None"""
+
+        self.build_inventory_hosts()
+        self.build_inventory_groups()
+        # import pdb; pdb. set_trace()
+
+    def _populate(self):
+        """Return the hosts and groups
+
+        Returns the processed container configurations from the lxd import
+
+        Args:
+            None
+        Kwargs:
+            None
+        Raises:
+            None
+        Returns:
+            None"""
+
+        if len(self.data) == 0:  # If no data is injected by unittests open socket
+            self.socket = self._connect_to_socket()
+            self.get_container_data(self._get_containers())
+            self.get_network_data(self._get_networks())
+
+        self.extract_information_from_container_configs()
+
+        if self.dump_data:
+            self.save_json_data(self.TEST_PATH)
```

Comment:
This looks like test code (in the middle of the actual code). If so, we should replace that with mock objects in `test_lxd.py` instead.

Reply:
I built in dump_data so that the user can save the data from lxd in order to attach them to a bug report. I am unsure whether there are better ways to achieve this goal in Ansible.

Reply:
The best way would be to have a `_info` module which queries this information. That could also be used in other contexts.